### DISCUSSION TITLE
Error running Peacock

### DISCUSSION THREAD

**Discussion by ajyano22** [idaholab/moose#17394] [Q&A General] [answered]
Hello,

I'm pretty new to Moose and Linux in general, I think I have everything set up correctly but I'm still getting errors when trying to run a simulation through peacock. This is the error:
moose) [aliyay@aliya-hpspectrex360convertible15ch0xx peacock]$ ./peacock
No executable found for method type(s): opt, dbg, oprof, devel
QXcbConnection: XCB error: 3 (BadWindow), sequence: 2300, resource id: 10515013, major code: 40 (TranslateCoords), minor code: 0
QXcbConnection: XCB error: 3 (BadWindow), sequence: 2904, resource id: 10517754, major code: 40 (TranslateCoords), minor code: 0
QXcbConnection: XCB error: 3 (BadWindow), sequence: 3590, resource id: 10519280, major code: 40 (TranslateCoords), minor code: 0
QXcbConnection: XCB error: 3 (BadWindow), sequence: 3953, resource id: 10526005, major code: 40 (TranslateCoords), minor code: 0
WARNING
 Problem running '/home/aliyay/projects/moose/modules/combined/practice/workshop2/model_A_equilibrium.i -options_left 0 --json'
WARNING
 Failed to load json from '/home/aliyay/projects/moose/modules/combined/practice/workshop2/model_A_equilibrium.i': Problem running '/home/aliyay/projects/moose/modules/combined/practice/workshop2/model_A_equilibrium.i -options_left 0 --json'
Error: [Errno 13] Permission denied: '/home/aliyay/projects/moose/modules/combined/practice/workshop2/model_A_equilibrium.i'

This says the issue is the permissions but when I check them, it says I should be able to execute these files.
(moose) [aliyay@aliya-hpspectrex360convertible15ch0xx workshop2]$ ls -la
total 124
drwxr-xr-x 3 aliyay aliyay  4096 Mar 17 11:43 .
drwxr-xr-x 4 aliyay aliyay  4096 Mar 17 11:26 ..
drwx------ 2 aliyay aliyay  4096 Mar 17 11:42 .jitcache
-rw-r--r-- 1 aliyay aliyay    74 Mar 17 11:43 model_A_eq_1D.csv
-rw-r--r-- 1 aliyay aliyay 85384 Mar 17 11:43 model_A_eq_1D.e
-rw-r--r-- 1 aliyay aliyay    13 Mar 17 11:42 model_A_eq_1D_x_direction_0000.csv
-rw-r--r-- 1 aliyay aliyay  2494 Mar 17 11:43 model_A_eq_1D_x_direction_0001.csv
-rwxrwxrwx 1 aliyay aliyay  2704 Mar 17 11:23 model_A_equilibrium.i
-rwxrwxrwx 1 aliyay aliyay  2620 Mar 17 11:23 model_A_growth.i
-rwxrwxrwx 1 aliyay aliyay  2577 Mar 17 11:23 model_A_shrinkage.i
-rwxrwxrwx 1 aliyay aliyay  2658 Mar 17 11:23 model_A_shrinkage_n_growth.i

 I'm not sure what the issue is. Any help is appreciated.

**Comment by aeslaughter** [comment]
You have not compiled your code:  "No executable found for method type(s): opt, dbg, oprof, devel". If you are new I suggest working through out tutorial. https://mooseframework.inl.gov/getting_started/examples_and_tutorials/tutorial01_app_development/index.html

**Comment by milljm** [comment, marked as answer]
The error lies in line right after you executed peacock:

`No executable found for method type(s): opt, dbg, oprof, devel`

While you may have built modules, or the the combined module in particular, Peackock does not know how to search that direction. Peacock will search for binaries at your current working directory, and work 'backwards' (parent directory):

```bash
/home/aliyay/projects/moose/python/peacock
  nope. Go up one directory.
/home/aliyay/projects/moose/python
  nope. Go up one directory.
/home/aliyay/projects/moose
  nope. Go up one directory. Woops. We hit the root directory, tell the user:
`No executable found for method type(s): opt, dbg, oprof, devel`
```

What you need to do, is build the combined module, and then run peacock from either that location, or from the location of your chosen input file (because combined-opt will exist in a parent directory from that location):

```bash
cd /home/aliyay/projects/moose/modules/combined
make     # if you haven't done so already
cd /home/aliyay/projects/moose/modules/combined/practice/workshop2
/home/aliyay/projects/moose/python/peacock/pecock -i model_A_equilibrium.i
```

That's a lot of paths... Which is why we suggest created a `PATH` to Peacock so that you can just execute `peacock` no matter where you are along the file system:

```bash
export PATH=/home/aliyay/projects/moose/python/peaock:$PATH
cd /home/aliyay/projects/moose/modules/combined/practice/workshop2
peacock
```
(of course, the same applies: you must have a binary built somewhere in your parent directories).

I would also suggest throwing that `export PATH=/home/aliyay/projects/moose/python/peaock:$PATH` in your bash profile, so it is always set.
